androidClient/app/src/main/java/com/macbitsgoa/comrades/coursematerial/DownloadFile.java

```diff
@@ -0,0 +1,157 @@
+package com.macbitsgoa.comrades.coursematerial;
+
+import android.app.PendingIntent;
+import android.content.Context;
+import android.content.Intent;
+import android.net.Uri;
+import android.os.AsyncTask;
+import android.util.Log;
+import android.widget.Toast;
+
+import com.macbitsgoa.comrades.R;
+
+import java.io.BufferedInputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.URL;
+import java.net.URLConnection;
+
+import androidx.core.app.NotificationCompat;
+import androidx.core.app.NotificationManagerCompat;
+
+import static android.os.Environment.getExternalStorageDirectory;
+import static com.macbitsgoa.comrades.CHC.TAG_PREFIX;
+
+public class DownloadFile extends AsyncTask<Void, Integer, Integer> {
```

Comment:
Can we use Workmanager? https://developer.android.com/topic/libraries/architecture/workmanager#java

Reply:
any advantage of using this? Downloads should be placed immediately when user wants.so why schedule them?

Reply:
Async task has disadvantage of causing leaks. And, google is changing ways of handling background tasks again and again with newer apis. The work manager seems to handle this automatically.